sdcm/cluster_gce.py

```
@@ -382,12 +384,80 @@ def _create_instances(self, count, dc_idx=0, instance_type=None):
+                    self._gce_zone_names[dc_idx] = new_provisioner.availability_zone
+                    self.log.info("Successfully provisioned instances in fallback zone %s-%s", region, alt_zone)
+                    return result
+                except (ZoneResourcesExhaustedError, ProvisionError) as fallback_exc:
+                    if isinstance(fallback_exc, ProvisionError) and not _is_zone_exhausted(fallback_exc):
+                        raise
+                    self.log.warning("Fallback zone %s-%s also exhausted, trying next", region, alt_zone)
+                    continue
+            raise ZoneResourcesExhaustedError(
+                f"All zones in region {region} are exhausted: tried {exhausted_zone} and "
+                f"{[f'{region}-{z}' for z in alternative_zones]}"
+            )
```

Ruff (0.15.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)